### PR TITLE
Bug 1836022 - catch NS_ERROR_HOST_IS_IP_ADDRESS when querying cookie banners exception

### PR DESCRIPTION
**Pull request by jackyzy823**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1836022

### CODE REVIEW

File: android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/cookiebanners/GeckoCookieBannersStorage.kt [hunk 1]
```diff
@@ -105,8 +105,9 @@ class GeckoCookieBannersStorage(
                 )
             }
         } catch (e: Exception) {
-            if ((e.message ?: "").contains("NS_ERROR_INSUFFICIENT_DOMAIN_LEVELS")) {
-                // This normally happen on internal sites like about:config
+            // This normally happen on internal sites like about:config or ip sites.
+            val geckoError = listOf("NS_ERROR_INSUFFICIENT_DOMAIN_LEVELS", "NS_ERROR_HOST_IS_IP_ADDRESS")
+            if (geckoError.any { (e.message ?: "").contains(it) }) {
```
Comment (MatthewTighe) [inline android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/cookiebanners/GeckoCookieBannersStorage.kt:110]: nit:
```suggestion
            val disabledErrors = listOf("NS_ERROR_INSUFFICIENT_DOMAIN_LEVELS", "NS_ERROR_HOST_IS_IP_ADDRESS")
            if (disabledErrors.any { (e.message ?: "").contains(it) }) {
```

**Review by MatthewTighe — COMMENTED** [review]
Besides my nit, this LGTM and solves the crash. I'll cycle back for approval once the name is updated 🙂

**Comment by MatthewTighe** [comment]
I filed [a bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1837082) to see if we can get product input on further improvements here, but I think this patch will be worth landing as is in order to fix the crasher.

**Comment by jackyzy823** [comment]
@MatthewTighe  The `always off even manually enable` problem could be solved here 

change
https://github.com/mozilla-mobile/firefox-android/blob/dcc1820500e117c7fea94a0d7868a62235ea71d2/fenix/app/src/main/java/org/mozilla/fenix/settings/quicksettings/protections/cookiebanners/CookieBannersStorageExt.kt#L40-L43
to 
```kotlin
 if (hasException) { 
     CookieBannerUIMode.HIDE
 }
```
which I think is reasonable. If you feel ok , i will add it to commit.

**Review by MatthewTighe — APPROVED** [review]

**Comment by MatthewTighe** [comment]
@jackyzy823 This does seem like a reasonable approach, but I would prefer to let some UX or product people weigh in before we move forward with a solution. I'll add your suggestion to bug report I filed above, though, thanks!

Let's land this patch as is to address the crash, and then we can see about moving forward. Thanks again for another contribution!

**Comment by MatthewTighe** [comment]
bors try

**Pull request merged**